types/mapbox__mapbox-gl-draw/index.d.ts

```diff
@@ -0,0 +1,32 @@
+// Type definitions for @mapbox/mapbox-gl-draw 1.1
+// Project: https://github.com/mapbox/mapbox-gl-draw#readme
+// Definitions by: Ryan Vanbelkum <https://github.com/ryanvanbelkum>
+//                 Nathan Weber <https://github.com/nweber-gh>
+//                 Bryan Wain <https://github.com/bdwain>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+declare module "@mapbox/mapbox-gl-draw/src/constants" {
+  enum modes {
+    SIMPLE_SELECT,
+    DIRECT_SELECT,
+    DRAW_POINT,
+    DRAW_POLYGON,
+    DRAW_LINE_STRING,
+  }
+}
+
+declare module "@mapbox/mapbox-gl-draw" {
```

Comment:
These paths should come from the actual file structure, not ambient module declarations. So, create a `src/constants.d.ts` for the stuff above, and just move the contents of this block out to the root level (index.d.ts is already the correct filename for it.)